src/main/java/calcservlet/ResultFilter.java

```diff
@@ -0,0 +1,73 @@
+package calcservlet;
+
+import javax.servlet.*;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+import java.util.StringTokenizer;
+
+@WebFilter(filterName = "ResultFilter", urlPatterns = "/calc/result")
+public class ResultFilter implements Filter {
+    public void destroy() {
+    }
+
+    public void doFilter(ServletRequest req, ServletResponse resp, FilterChain chain) throws ServletException, IOException {
+        HttpServletRequest request = (HttpServletRequest) req;
+        HttpServletResponse response = (HttpServletResponse) resp;
+        HttpSession session = request.getSession();
+        if(session.getAttribute("equation") != null) {
+            String qtn = session.getAttribute("equation").toString();
+            if (!goodFormat(qtn)){
+                response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            } else if (!hasVars(session)) {
+                response.setStatus(HttpServletResponse.SC_CONFLICT);
+            } else {
+                chain.doFilter(req, resp);
+            }
+        }
+
+    }
+
+    public void init(FilterConfig config) {
+
+    }
+
+    private boolean goodFormat(String qtn) {
+        StringTokenizer tokenizer = new StringTokenizer(qtn, "() ");
+        StringBuilder builder = new StringBuilder();
+        String q;
+        while (tokenizer.hasMoreElements()) {
+            builder.append(tokenizer.nextElement());
+        }
+        q = builder.toString();
+        for(int i = 0; i < q.length()-1; i+=2) {
+            if(Character.isAlphabetic(q.charAt(i)) || Character.isDigit(q.charAt(i)) &&  "+-*/".contains(String.valueOf(q.charAt(i+1)))) {
+                continue;
+            }
+            return false;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid using a branching statement as the last in a loop.](https://app.codacy.com/app/thejerome/IFMO_Servlets_20190915/pullRequest?prid=4510206)